### ISSUE TITLE
Command which just presses a keyboard button

### ISSUE THREAD

**Issue by magnus-ISU** [DFHack/dfhack#2438]
Motivation: adding player-defined hotkeys is great.

But I want a hotkey to create a manager order in one press instead of 3, or to open food screen in one press instead of 3; there are any number of tasks which are all so common that they should be default, but they aren't.

DF's built-in macro system somehow cannot assign macros to hotkeys. As far as I can tell, `keybinding` can't replace that, because the above such command doesn't exist; but I hope it is easy to implement and thus should be.

**Comment by myk002** [comment]
It is tricky to send macro-like sequences to DF since it doesn't read keys. It reads DF key code keybinding tokens, like what you see in `data/init/interface.txt`. luckily, we already have a way to do that transformation - quickfort. Unlucky, quickfort assumes it is running map-oriented blueprints, and has safety checks to ensure mis-aligned blueprints don't send keystrokes in "error" conditions, which is what the use case here would be detected as.

Proposal: add a new "macro" mode to quickfort, similar to config mode, but with no safety checks. The code is already structured to support this easily.

Then we can create a `macro` script which is just a thin wrapper around the quickfort API. The macro script will take a string in blueprint syntax (e.g. `!` or `{Ctrl}` for a control modifier key) and use quickfort to parse and apply the key code sequence.

In other words, yes, this will be easy to do.

In the meantime, you could synthesize a quick but less flexible solution by setting up a DFHack keybinding that runs something like `:lua simulateInput{KEYNAME=true}` and hard code that for the macros you want.

**Comment by lethosor** [comment]
There is also the devel/send-key script, which can be used to send individual keys. It sounds like you want to send a sequence.

**Comment by magnus-ISU** [comment]
Send-key might be what I want. A command already exists to do multiple commands in a sequence
